packages/nova-react/src/graphql/hooks.test.tsx

```diff
@@ -152,9 +156,87 @@ describe(useFragment, () => {
       useFragment(fragment, opaqueFragmentRef);
     void _;
   });
+
+  it("allows null or undefined to be passed as the fragment ref", () => {
+    const fragment = {} as unknown as GraphQLTaggedNode;
+
+    const { result, rerender } = renderHook<
+      { fragmentRef: null | undefined },
+      null | undefined
+    >(({ fragmentRef }) => useFragment(fragment, fragmentRef), {
+      wrapper: ({ children }) => (
+        <NovaGraphQLProvider graphql={{}}>{children}</NovaGraphQLProvider>
+      ),
+      initialProps: {
+        fragmentRef: null,
+      },
+    });
+
+    expect(result.current).toBeNull();
+
+    rerender({ fragmentRef: undefined });
+
+    expect(result.current).toBeUndefined();
+  });
+
+  it("return type can be `null`, `undefined` or data", () => {
+    type SomeFragment$data = { someKey: string };
+    type SomeFragment$key = {
+      readonly " $data"?: SomeFragment$data;
+      readonly " $fragmentRefs": FragmentRefs<"SomeFragment">;
+    };
+
+    const fragment = {} as unknown as GraphQLTaggedNode;
+    const opaqueFragmentRef = {} as unknown as SomeFragment$key | null;
+
+    () => {
```

Comment:
You can start a new scope with just the curly braces, which might make it slightly less confusing—I spend a few sec trying to understand what you were trying to do with the function.

No biggie, though.

Reply:
I used the function to not actually call `useFragment` which will give the invalid hook error ;) So seems that wasn't actually clear after all, I'll fix it.